File: CS50/pset6/readability.py
from cs50 import get_string
from sys import exit
from decimal import *

def main():
    texto = get_string("Digite o texto: ")
    if len(texto) <= 1:
        print("Invalid text")
        exit(1)

    sentences = 0
    letras = 0
    palavras = 1
    for i in range(len(texto)):
        if texto[i] == '!' or texto[i] == '.' or texto[i] == '?':
            sentences += 1

        elif texto[i] == " ":
            palavras += 1

        elif texto[i].upper() >= 'A' and texto[i].upper() <= 'Z':
            letras += 1

    s = Decimal(sentences) / Decimal(palavras) * 100
    l = Decimal(letras) / Decimal(palavras) * 100
    index = (Decimal(0.0588) * l) - (Decimal(0.296) * s) - Decimal(15.8)

    if index <= 1:
        print("Before Grade 1")
        exit(0)

    elif index > 16:
        print("Grade 16+")
        exit(0)

    else:
        print(f"Grade: {round(index)}")
        exit(0)

main()

File: CS50/pset1/readability.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>

int main(void)
{
    char *text = get_string("\rDigite o texto: ");
    int letters = 0;
    int sentences = 0;
    int palavras = 1;
    for (int count = 0; count < strlen(text); count++)
    {
        if(text[count] >= 65 && text[count] <= 90)
            letters++;

        else if((text[count]) >= 97 && text[count] <= 122)
            letters++;

        else if((text[count]) == 32)
            palavras++;

        else if((text[count]) == 33 || text[count] == 46 || text[count] == 63)
            sentences++;
    }
    double index = ((0.0588 * ((float)letters / palavras * 100.0)) - (0.296 * ((float)sentences / palavras * 100.0)) - 15.8 + 0.4);
    if (index < 1)
     printf("\rBefore grade 1 %d\n", (int)index);

    else
     printf("\rGrade %d\n", (int)index);

    return 0;
}